recipes/libsigcpp/2.x.x/conanfile.py

```diff
@@ -0,0 +1,114 @@
+import os
+
+from conans.errors import ConanInvalidConfiguration
+from conans import ConanFile, Meson, tools
+
+required_conan_version = ">=1.29.0"
+
+class LibSigCppConan(ConanFile):
+    name = "libsigcpp"
+    homepage = "https://github.com/libsigcplusplus/libsigcplusplus"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "LGPL-3.0"
+    description = (
+        "libsigc++ implements a typesafe callback system for standard C++."
+    )
+    topics = ("conan", "libsigcpp", "callback")
+    settings = "os", "compiler", "arch", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {"shared": False, "fPIC": True}
+
+    _meson = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def build_requirements(self):
+        self.build_requires("meson/0.56.0")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if self.settings.compiler == "Visual Studio":
+            self.options.shared = True
+
+    def configure(self):
+        compiler = str(self.settings.compiler)
+        version = tools.Version(self.settings.compiler.version)
+        if (
+            not self.options.shared
+            and compiler == "Visual Studio"
+            and version <= "16"
+        ):
+            raise ConanInvalidConfiguration(
+                "{} {} is not supported for static compilation".format(
+                    compiler,
+                    version,
+                )
+            )
+
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("libsigc++-{}".format(self.version), self._source_subfolder)
+        tools.replace_in_file(
+            os.path.join(
+                self.source_folder, self._source_subfolder, "meson.build"
+            ),
+            "subdir('tests')",
+            "",
+        )
+
+    def _configure_build(self):
+        if self._meson:
+            return self._meson
+        self._meson = Meson(self)
+
+        args = [
```

Comment:
From what I understand, next time you can do https://github.com/conan-io/conan-center-index/blob/3335a0223cd3593bd8d10b2789aac55d564b3ef3/recipes/libpsl/all/conanfile.py#L77